### PR TITLE
Added HashPuzzleP2PKH (Sha256) example contract & tests.

### PR DESCRIPTION
**Pull request by freddiehonohan**
This is a simple script which allows people to embed data verifiable by it's sha256sum into transaction input scripts.

### CODE REVIEW

**Comment by freddiehonohan** [comment]
This contract was written to highlight the insecurity of segregating the input data from the calculation of the transaction hash also known as segregated witness.

File: tests/js/verifydatap2pkh.scrypttest.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+const data =  dataBuffer
+const sha256Data = bsv.crypto.Hash.sha256(dataBuffer);
+
+describe('Test sCrypt contract VerifyDataP2PKH In Javascript', () => {
+  let demo
+  let sig
+
+  before(() => {
+    const VerifyDataP2PKH = buildContractClass(path.join(__dirname, '../../contracts/verifydatap2pkh.scrypt'), tx, inputIndex, inputSatoshis)
+    demo = new VerifyDataP2PKH(toHex(pkh), toHex(sha256Data))
```
Comment (xhliu) [inline tests/js/verifydatap2pkh.scrypttest.js]: Call it verifyDataP2PKH, following naming convention.

File: tests/js/verifydatap2pkh.scrypttest.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    sig = signTx(tx, privateKey, demo.getScriptPubKey())
+    expect(demo.verify(toHex(data), toHex(sig),  toHex(publicKey))).to.equal(true);
+  });
+
+  it('signature check should fail when wrong private key signs', () => {
+    sig = signTx(tx, privateKey2, demo.getScriptPubKey())
+    expect(demo.verify(toHex(data), toHex(sig),  toHex(publicKey))).to.equal(false);
+  });
+
+  it('sha256 check should succeed when right data is provided', () => {
```
Comment (xhliu) [inline tests/js/verifydatap2pkh.scrypttest.js]: Duplicate with the first test case, when both right data and private key are provided.

File: contracts/verifydatap2pkh.scrypt [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    // Public Key Hash and Sha256Sum are provided to the contract
+    constructor(Ripemd160 pubKeyHash, Sha256 shaSum) {
+        this.pubKeyHash = pubKeyHash;
+        this.hash = shaSum;
+    }
+
+    // The Data is verified to match the Sha256Sum
+    function verifyChecksum(bytes dataBytes) returns(bool){
+        return(sha256(dataBytes) == this.hash);
```
Comment (xhliu) [inline contracts/verifydatap2pkh.scrypt]: This kind of contract is called a hash puzzle contract. See `HashPuzzle` https://scryptdoc.readthedocs.io/en/latest/contracts.html#multiple-contracts. U can use a standard contract `HashPuzzleSha256` for this as u did for `P2PKH`.

File: tests/testnet/hashpuzzlep2pkh.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+const publicKey = privateKey.publicKey
+const pkh = bsv.crypto.Hash.sha256ripemd160(publicKey.toBuffer())
+
+// NIST Test Vectors (https://www.nist.gov/itl/ssd/software-quality-group/nsrl-test-data)
+const dataBuffer = Buffer.from("abc");
+const data =  dataBuffer
+const sha256Data = bsv.crypto.Hash.sha256(dataBuffer);
+
+// private key on testnet in WIF
+const key = '92AiMWahQ4pgfiix7s2Wiyx4vwSB2AQRsyqZC1LfsV8tyEUeHU4'
```
Comment (xhliu) [inline tests/testnet/hashpuzzlep2pkh.js]: Remove this.

**Pull request merged**